libs/utils/trace.py

```diff
@@ -728,6 +734,77 @@ def _loadFunctionsStats(self, path='trace.stats'):
 
         return len(self._functions_stats_df) > 0
 
+    @memoized
+    def getCPUActiveSignal(self, cpu):
+        """
+        Build a square wave representing the active (i.e. non-idle) CPU time,
+        i.e.:
+            cpu_active[t] == 1 if at least one CPU is reported to be
+                               non-idle by CPUFreq at time t
+            cpu_active[t] == 0 otherwise
+
+        :param cpu: CPU ID
+        :type cpu: int
+
+        :returns: :mod:`pandas.Series`
+        """
+        if not self.hasEvents('cpu_idle'):
+            logging.warn('Events [cpu_idle] not found, '\
+                         'cannot compute CPU active signal!')
+            return None
+
+        idle_df = self._dfg_trace_event('cpu_idle')
+        cpu_df = idle_df[idle_df.cpu_id == cpu]
+
+        cpu_active = cpu_df.state.apply(
+            lambda s: 1 if s == NON_IDLE_STATE else 0
+        )
+
+        start_time = 0.0
+        if not self.ftrace.normalized_time:
+            start_time = self.ftrace.basetime
+        if cpu_active.index[0] != start_time:
+            entry_0 = pd.Series(cpu_active.iloc[0] ^ 1, index=[start_time])
+            cpu_active = pd.concat([entry_0, cpu_active])
+
+        return cpu_active
+
+    @memoized
+    def getClusterActiveSignal(self, cluster):
+        """
+        Build a square wave representing the active (i.e. non-idle) cluster
+        time, i.e.:
+            cluster_active[t] == 1 if at least one CPU is reported to be
+                                   non-idle by CPUFreq at time t
+            cluster_active[t] == 0 otherwise
+
+        :param cluster: list of CPU IDs belonging to a cluster
+        :type cluster: list(int)
+
+        :returns: :mod:`pandas.Series`
+        """
+        if not self.hasEvents('cpu_idle'):
+            logging.warn('Events [cpu_idle] not found, '\
+                         'cannot compute cluster active signal!')
+            return None
+
+        cpu_active = {}
+        for cpu in cluster:
+            cpu_active[cpu] = self.getCPUActiveSignal(cpu)
+
+        active = pd.DataFrame(cpu_active)
```

Comment:
It seems that the culprit is the trace. I'll modify the code with your patch because apparently `pandas` accepts duplicate indexes if you provide it explicitly (i.e. you do something like `pd.DataFrame([1, 2, 3, 4], index=[1, 2, 2, 3])`) or if you do a `join` as you have seen with your solution.